purchase_order_variant_mgmt/wizard/purchase_manage_variant.py

```diff
@@ -0,0 +1,104 @@
+        self.variant_line_ids = [(6, 0, [])]
+        template = self.product_tmpl_id
+        context = self.env.context
+        record = self.env[context['active_model']].browse(
+            context['active_id'])
+        if context['active_model'] == 'purchase.order.line':
+            purchase_order = record.order_id
+        else:
+            purchase_order = record
+        num_attrs = len(template.attribute_line_ids)
+        if not template or not num_attrs:
+            return
+        line_x = template.attribute_line_ids[0]
+        line_y = False if num_attrs == 1 else template.attribute_line_ids[1]
+        lines = []
+        for value_x in line_x.value_ids:
+            for value_y in line_y and line_y.value_ids or [False]:
+                product = self._get_product_variant(value_x, value_y)
+                if not product:
+                    continue
+                order_line = purchase_order.order_line.filtered(
+                    lambda x: x.product_id == product
+                )[:1]
+                lines.append((0, 0, {
+                    'value_x': value_x,
+                    'value_y': value_y,
+                    'product_uom_qty': order_line.product_qty,
+                }))
+        self.variant_line_ids = lines
+
+    @api.multi
+    def button_transfer_to_order(self):
+        context = self.env.context
+        record = self.env[context['active_model']].browse(context['active_id'])
+        if context['active_model'] == 'purchase.order.line':
+            purchase_order = record.order_id
+        else:
+            purchase_order = record
+        OrderLine = self.env['purchase.order.line']
+        lines2unlink = OrderLine
+        for line in self.variant_line_ids:
+            product = self._get_product_variant(line.value_x, line.value_y)
+            order_line = purchase_order.order_line.filtered(
+                lambda x: x.product_id == product
+            )
+            if order_line:
+                if not line.product_uom_qty:
+                    # Done this way because there's a side effect removing here
+                    lines2unlink |= order_line
+                else:
+                    order_line.product_qty = line.product_uom_qty
+            elif line.product_uom_qty:
+                order_line = OrderLine.new({
+                    'product_id': product.id,
+                    'product_uom': product.uom_id,
+                    'product_uom_qty': line.product_uom_qty,
+                    'order_id': purchase_order.id,
+                })
+                order_line.onchange_product_id()
+                # This should be done later for handling supplier quantities
+                order_line.product_qty = line.product_uom_qty
+                order_line._onchange_quantity()
+                order_line_vals = order_line._convert_to_write(
+                    order_line._cache)
+                self.env['purchase.order.line'].create(order_line_vals)
```

Comment:
wouldn't be better to collect line values as `(0, 0, vals)` and write on the order once?